### PR TITLE
Checking if window.document defined before trying to resolve url

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (moozzyk) [inline .vscode/launch.json]: I had to copy this from my gist too many times

**Comment by moozzyk** [comment]
[Looks](https://github.com/aspnet/SignalR/issues/957#issuecomment-332648083) like when using React Native you can have `window` but without the `document` property.

File: client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HttpConnection.ts [hunk 2]
```diff
@@ -162,7 +162,7 @@ export class HttpConnection implements IConnection {
             return url;
         }
 
-        if (typeof window === 'undefined') {
+        if (typeof window === 'undefined' || !window || !window.document) {
```
Comment (davidfowl) [inline client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HttpConnection.ts]: do we need the undefined check

Comment (moozzyk) [reply client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HttpConnection.ts]: It's more idiomatic. I guess `!window` will would work for both `null` and `undefined` but the semantic is a bit different.

**Pull request merged**